### PR TITLE
Make containers readonlyrootFileSystem=true compatible

### PR DESCRIPTION
**Pull request by sir-sigurd**
## Description
- s3-proxy: adjust some paths to be aligned with other nginx containers
- write runtime generated configs to /tmp

### CODE REVIEW

File: catalog/Dockerfile [hunk 1]
```diff
@@ -16,18 +16,23 @@ RUN ln -sf /dev/stdout /var/log/nginx/access.log && \
...
 # Copy pre-built catalog assets to nginx
-RUN rm -rf /usr/share/nginx/html
-COPY build /usr/share/nginx/html
+RUN rm -rf $NGINX_STATIC_DIR
+COPY build $NGINX_STATIC_DIR
 
 # Copy config file
 COPY config.json.tmpl config.json.tmpl
 
+# XXX: use subdir in /tmp?
```
Comment (nl0) [inline catalog/Dockerfile:27]: what for? do you foresee any problems here?

**Review by nl0 — APPROVED** [review]
lgtm

File: catalog/Dockerfile (hunk 1 above)
Comment (sir-sigurd) [reply catalog/Dockerfile:27]: it kinda feels safer, but I don't think it's really needed

Comment (sir-sigurd) [inline catalog/Dockerfile:27]:
```suggestion
```

**Review by nl0 — APPROVED** [review]

**Pull request merged**